File: src/main/java/com/nunezdev/inventory_manager/service/UserService.java
package com.nunezdev.inventory_manager.service;


import java.util.List;

import org.springframework.stereotype.Service;

import com.nunezdev.inventory_manager.dto.UserDTO;

@Service
public interface UserService {
    boolean verifyLogin(UserDTO userDTO);
    UserDTO createUser(UserDTO userDTO);
    List<UserDTO> getAllUsers();
    boolean updateUserRole(Long userId, String role);
    boolean updateUser(UserDTO userDTO);
    Long getUserIdByUsername(String username);
}

File: src/main/capstone_angular/src/app/service/user/UserService.ts
import {Observable} from "rxjs";
import {HttpClient} from "@angular/common/http";
import {Injectable} from "@angular/core";
import {Location, LocationStrategy} from "@angular/common";
import {User} from "../../model/user.model";
import {UserDTO} from "../../model/UserDTO.model";


@Injectable({
  providedIn: 'root'
})

export class UserService {
  private userUrl: string = 'http://localhost:8080/api/users';
  //private userUrl;

  constructor(private http: HttpClient, private location: Location) {
    //this.userUrl = this.location.prepareExternalUrl('/api/users');
  }

  createUser(userData: any): Observable<any> {
    return this.http.post(`${this.userUrl}/register`, userData);
  }

  // New method to fetch all users
  getAllUsers(): Observable<UserDTO[]> {
    return this.http.get<UserDTO[]>(`${this.userUrl}`);
  }

  // New method to update a user's role
  updateUserRole(userId: number, role: string): Observable<any> {
    return this.http.put(`${this.userUrl}/${userId}/role`, { role });
  }

  updateUser(userData: UserDTO): Observable<any> {
    return this.http.put(`${this.userUrl}/${userData.id}`, userData);
  }

}
